app/assets/javascripts/discourse/tests/acceptance/composer-form-template-test.js

```diff
@@ -0,0 +1,130 @@
+import { click, fillIn, visit } from "@ember/test-helpers";
+import { toggleCheckDraftPopup } from "discourse/services/composer";
+import { cloneJSON } from "discourse-common/lib/object";
+import TopicFixtures from "discourse/tests/fixtures/topic";
+import { acceptance } from "discourse/tests/helpers/qunit-helpers";
+import selectKit from "discourse/tests/helpers/select-kit-helper";
+import { test } from "qunit";
+
+acceptance("Composer Form Template", function (needs) {
+  needs.user({
+    id: 5,
+    username: "kris",
+    whisperer: true,
+  });
+  needs.settings({
+    experimental_form_templates: true,
+    general_category_id: 1,
+    default_composer_category: 1,
+  });
+  needs.site({
+    can_tag_topics: true,
+    categories: [
+      {
+        id: 1,
+        name: "General",
+        slug: "general",
+        permission: 1,
+        topic_template: null,
+        form_template_ids: [1],
+      },
+      {
+        id: 2,
+        name: "test too",
+        slug: "test-too",
+        permission: 1,
+        topic_template: "",
+      },
+    ],
+  });
+  needs.pretender((server, helper) => {
+    server.put("/u/kris.json", () => helper.response({ user: {} }));
+
+    server.get("/form-templates/1.json", () => {
+      return helper.response({
+        form_template: {
+          name: "Testing",
+          template: `- type: input
+  id: full-name
+  attributes:
+    label: "Full name"
+  description: "What is your full name?"
+- type: textarea
+  id: description
+  attributes:
+    label: "Description"`,
+        },
+      });
+    });
+
+    server.get("/posts/419", () => {
+      return helper.response({ id: 419 });
+    });
+    server.get("/composer/mentions", () => {
+      return helper.response({
+        users: [],
+        user_reasons: {},
+        groups: { staff: { user_count: 30 } },
+        group_reasons: {},
+        max_users_notified_per_group_mention: 100,
+      });
+    });
+    server.get("/t/960.json", () => {
+      const topicList = cloneJSON(TopicFixtures["/t/9/1.json"]);
+      topicList.post_stream.posts[2].post_type = 4;
+      return helper.response(topicList);
+    });
+  });
+
+  needs.hooks.afterEach(() => toggleCheckDraftPopup(false));
+
+  test("Composer Form Template is shrank and reopened", async function (assert) {
+    await visit("/");
+    await click("#create-topic");
+
+    assert.strictEqual(selectKit(".category-chooser").header().value(), "1");
+
+    assert.strictEqual(
+      document.documentElement.style.getPropertyValue("--composer-height"),
+      "var(--new-topic-composer-height, 400px)",
+      "sets --composer-height to 400px when creating topic"
+    );
```

Comment:
Maybe check on the `#reply-control` element classes here? The composer should have a class of `open` set when open and `draft` or `saving` when in the minimized state.

Reply:
Thanks @pmusaraj! I had used the same composer-height checks from the composer tests, but I changed it now to use the `open`/`draft` classes instead of relying on hardcoded heights.